source/common/upstream/load_balancer_impl.h

```diff
@@ -518,6 +518,7 @@ class EdfLoadBalancerBase : public ZoneAwareLoadBalancerBase {
 
   // Upstream::ZoneAwareLoadBalancerBase
   HostConstSharedPtr peekAnotherHost(LoadBalancerContext* context) override;
+  HostConstVectorSharedPtr getallHosts(LoadBalancerContext* ) override {return nullptr;}
```

Comment:
i have declared this as a PURE virtual function, so it has to be implemented by any of its derived class.
but nevertheless , i will check if this can be removed from PURE and selectively implement only in our class.
also i am think of implementing a getter to get this thread local data or also gettign it directly from TSL(thread local storage if possible)